### PR TITLE
api: support basic python extension

### PR DESCRIPTION
**Pull request by MingjiHan99**
Support basic Python API on x86_64 Linux and arm64 macOS

### CODE REVIEW

File: docs/07-python-extension.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Python Extension
+
+We are working to support Python API for user to process and analyze data in Python.  
+
+
+## Build and Use Python APIs
+```
+matruin build
+pip3 install ./target/wheels/package_name.whl
```
Comment (wangrunji0408) [inline docs/07-python-extension.md:9]: The command which works for me:
```suggestion
pip3 install maturin
matruin build
pip3 install ./target/wheels/risinglight-*.whl
```

File: docs/07-python-extension.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+python3
+import risinglight
+db = risnglight.open("risinglight.db")
```
Comment (wangrunji0408) [inline docs/07-python-extension.md:12]: typo
```suggestion
db = risinglight.open("risinglight.db")
```

File: src/lib.rs [hunk 3]
```diff
@@ -78,3 +78,88 @@ pub use self::db::{Database, Error};
...
 static GLOBAL: Jemalloc = Jemalloc;
+
+use std::path::PathBuf;
+
+/// Python Extension
+use pyo3::{prelude::*, wrap_pyfunction};
+use storage::SecondaryStorageOptions;
+/// Open a database for user, user can specify the path of database file
+#[pyfunction]
+fn open(path: String) -> PyResult<PythonDatabase> {
```
Comment (wangrunji0408) [inline src/lib.rs:89]: Could we move these code into a new file?

**Review by wangrunji0408 — COMMENTED** [review]
Great work! 🚀

File: src/lib.rs [hunk 4]
```diff
@@ -78,3 +78,88 @@ pub use self::db::{Database, Error};
...
+                let s = match array.get(row) {
+                    DataValue::Null => "null".to_string().to_object(py),
+                    DataValue::Bool(v) => v.to_object(py),
+                    DataValue::Int32(v) => v.to_object(py),
+                    DataValue::Int64(v) => (v).to_object(py),
+                    DataValue::Float64(v) => v.to_object(py),
+                    DataValue::String(s) if s.is_empty() => "(empty)".to_string().to_object(py),
+                    DataValue::String(s) => s.to_object(py),
+                    DataValue::Blob(s) if s.is_empty() => "(empty)".to_string().to_object(py),
+                    DataValue::Blob(s) => s.to_string().to_object(py),
```
Comment (wangrunji0408) [inline src/lib.rs:154]: I guess we don't need to deal with empty strings here.
```suggestion
                    DataValue::String(s) => s.to_object(py),
                    DataValue::Blob(s) => s.to_string().to_object(py),
```

File: src/lib.rs [hunk 5]
```diff
@@ -78,3 +78,88 @@ pub use self::db::{Database, Error};
...
+
+use std::path::PathBuf;
+
+/// Python Extension
+use pyo3::{prelude::*, wrap_pyfunction};
+use storage::SecondaryStorageOptions;
+/// Open a database for user, user can specify the path of database file
+#[pyfunction]
+fn open(path: String) -> PyResult<PythonDatabase> {
+    let runtime = tokio::runtime::Builder::new_current_thread()
```
Comment (wangrunji0408) [inline src/lib.rs:90]: Is it ok to use multiple threads here?
```suggestion
    let runtime = tokio::runtime::Builder::new_multi_thread()
```

Comment (MingjiHan99) [reply src/lib.rs:90]: Sure

File: src/lib.rs (hunk 3 above)
Comment (MingjiHan99) [reply src/lib.rs:89]: I don't know. Let me have a try.

**Review by wangrunji0408 — APPROVED** [review]

**Pull request merged**